### PR TITLE
Represent a template

### PR DESCRIPTION
**Pull request by AidanDelaney**
Create a template and the mechanism to ask questions form template prompts

Signed-off-by: Aidan Delaney <adelaney21@bloomberg.net>

### CODE REVIEW

**Review by samj1912 — APPROVED** [review]

File: pkg/internal/init_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package internal_test
+
+import (
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+)
+
+func TestIternal(t *testing.T) {
```
Comment (dfreilich) [inline pkg/internal/init_test.go:10]: `TestInternal`?

File: pkg/internal/template_test.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+	h "github.com/buildpacks/pack/testhelpers"
+	pseudotty "github.com/creack/pty"
+	"github.com/hinshun/vt10x"
+	"github.com/sclevine/spec"
+)
+
+func testReadPrompt(t *testing.T, when spec.G, it spec.S) {
+	when("Reading a prompt file", func() {
+		it("reads a correct prompt file", func() {
+			tmpDir, _ := ioutil.TempDir("", "test")
```
Comment (dfreilich) [inline pkg/internal/template_test.go:23]: nit: You can use `t.TempDir()` to simplify this, and not require the explicit `os.RemoveAll`

File: pkg/internal/template_test.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+			defer os.RemoveAll(tmpDir)
+			promptFile := filepath.Join(tmpDir, internal.PromptFile)
+			correctPromptFile := "[[prompt]]\nname=\"Foo\"\nprompt=\"Choose a foo\""
+			f, _ := os.Create(promptFile)
+			f.Write([]byte(correctPromptFile))
+			f.Close()
```
Comment (dfreilich) [inline pkg/internal/template_test.go:29]: nit: You can use `os.WriteFile()` to simplify this, if you're anyways closing the file descriptor here

**Review by dfreilich — COMMENTED** [review]
Looks great, and glad it was merged! Just a few small nits

**Pull request merged**